src/contexts/DatasetManagerContext.js

```diff
@@ -1,22 +1,50 @@
-// TEMPORARY
-// NOTE: incoporates the refinbio API client to manage the dataset
 import { createContext, useMemo } from 'react'
-import { useLocalStorage } from 'hooks/useLocalStorage'
+import { useRefinebio } from 'hooks/useRefinebio'
 
 export const DatasetManagerContext = createContext({})
 
 export const DatasetManagerContextProvider = ({ children }) => {
-  const [dataset, setDataset] = useLocalStorage('dataset', null) // TEMPORARY
-  const [datasetId, setDatasetId] = useLocalStorage('datasetId', null)
+  const {
+    dataset: datasetState,
+    setDataset: setDatasetState,
+    datasetId: datasetIdState,
+    downloadOptions: downloadOptionsState,
+    setDownloadOptions: setDownloadOptionsState,
+    setDatasetId: setDatasetIdState,
+    email: emailState,
+    token: tokenState
+  } = useRefinebio()
+
+  const dataset = datasetState
+  const setDataset = setDatasetState
+  const datasetId = datasetIdState
+  const downloadOptions = downloadOptionsState
+  const setDownloadOptions = setDownloadOptionsState
+  const setDatasetId = setDatasetIdState
+  const email = emailState
+  const token = tokenState
 
   const value = useMemo(
```

Comment:
The `dataset` stores the latest state of dataset, thus it's expensive. Let's keep it for this one, but I also agree with your point. Please see the merged [contexts](https://github.com/AlexsLemonade/refinebio-web/tree/dev/src/contexts) in the latest `dev`. All of them currently utilize the `useMemo` hook. 

Let's create a new issue to clean up the  `useMemo` hook from the following contexts 👍:
- `Band Context`
- `ModalContext`
- `SamplesTableManagerContext`
- `SearchManagerContext`

(* also clean up the no longer used `FilterContext` and its hook `useFilter` - replaced with `SearchManagerContext`)

**Side Note:** Our current `eslint` setting enforces the use of `useMemo` for the value object in Context, otherwise the exception comment must be added.

Reply:
I will create an issue for this.